lightning/src/ln/interactivetxs.rs

```diff
@@ -1149,19 +1219,68 @@ pub struct LocalOrRemoteInput {
 	prev_output: TxOut,
 }
 
+impl Writeable for LocalOrRemoteInput {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(1, self.serial_id, required),
+			(3, self.input.sequence.0, required),
+			(5, self.input.previous_output, required),
+			(7, self.input.script_sig, required),
+			(9, self.input.witness, required),
+			(11, self.prev_output, required),
+		});
+		Ok(())
+	}
+}
+
+impl Readable for LocalOrRemoteInput {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut serial_id: u64 = 0;
+		let mut sequence: u32 = 0;
+		let mut previous_output = OutPoint::new(Txid::from_byte_array([0; 32]), 0);
+		let mut script_sig = ScriptBuf::default();
+		let mut witness = Witness::new();
+		let mut prev_output =
+			TxOut { value: Amount::default(), script_pubkey: ScriptBuf::default() };
+		read_tlv_fields!(reader, {
+			(1, serial_id, required),
+			(3, sequence, required),
+			(5, previous_output, required),
+			(7, script_sig, required),
+			(9, witness, required),
+			(11, prev_output, required),
+		});
+		Ok(LocalOrRemoteInput {
+			serial_id,
+			input: TxIn { previous_output, script_sig, sequence: Sequence(sequence), witness },
+			prev_output,
+		})
+	}
+}
+
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub(crate) enum InteractiveTxInput {
 	Local(LocalOrRemoteInput),
 	Remote(LocalOrRemoteInput),
 	// TODO(splicing) SharedInput should be added
 }
 
+impl_writeable_tlv_based_enum!(InteractiveTxInput,
+	{1, Local} => (),
+	{3, Remote} => (),
```

Comment:
nit: typically we use:

```rust
(1, Local) => {},
(3, Remote) => {},
```

Also, can number these `0` and `1`.

Reply:
In this case these are both tuple variants, so the () => {} syntax doesn't work as it won't include the the `0` field of the tuple, so it doesn't compile. That syntax would only work for unit/struct variants, I believe.